merino/jobs/navigational_suggestions/domain_metadata_uploader.py

```diff
@@ -41,31 +39,57 @@ def __init__(
     def upload_top_picks(self, top_picks: str) -> Blob:
         """Upload the top pick contents to gcs."""
         bucket = self.storage_client.bucket(self.bucket_name)
-        dst_top_pick_name = self._destination_top_pick_name()
+        dst_top_pick_name = DomainMetadataUploader._destination_top_pick_name(
+            suffix=DomainMetadataUploader.DESTINATION_TOP_PICK_FILE_NAME_SUFFIX
+        )
+        self.update_latest_filename_suffix(
+            bucket_name=self.bucket_name,
+            file_suffix=DomainMetadataUploader.DESTINATION_TOP_PICK_FILE_NAME_SUFFIX,
+            bucket=bucket,
+            storage_client=self.storage_client,
+        )
         dst_blob = bucket.blob(dst_top_pick_name)
         dst_blob.upload_from_string(top_picks)
         return dst_blob
 
-    def _destination_top_pick_name(self) -> str:
+    @staticmethod
+    def _destination_top_pick_name(suffix: str) -> str:
         """Return the name of the top pick file to be used for uploading to GCS"""
         current = datetime.datetime.now()
-        return (
-            str(time.mktime(current.timetuple()) * 1000)
-            + "_"
-            + self.DESTINATION_TOP_PICK_FILE_NAME_SUFFIX
-        )
+        return f"{str(int(current.timestamp()))}_{suffix}"
```

Comment:
The timestamp doesn't have to be string-casted:

```suggestion
        return f"{int(current.timestamp())}_{suffix}"
```